In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('Train.csv')

In [3]:
data.sample(5)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
6562           FDD39        16.70          Low Fat         0.070441   
5954           DRI59          NaN          Low Fat         0.040637   
1058           FDH33        12.85          Low Fat         0.122221   
5633           DRC36        13.00              reg         0.044976   
8487           DRG13          NaN          Low Fat         0.037006   

        Item_Type  Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
6562        Dairy  216.8850            OUT018                       2009   
5954  Hard Drinks  224.6088            OUT027                       1985   
1058  Snack Foods   45.9428            OUT018                       2009   
5633  Soft Drinks  176.2054            OUT035                       2004   
8487  Soft Drinks  164.7526            OUT027                       1985   

     Outlet_Size Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
6562      Medium               Tier 3  Supermarket Type2          6275.1650  
5954      Medium               Tier 3  Supermarket Type3          4474.1760  
1058      Medium               Tier 3  Supermarket Type2          1274.3412  
5633       Small               Tier 2  Supermarket Type1          1225.7378  
8487      Medium               Tier 3  Supermarket Type3          4111.3150

In [4]:
data.shape

(8523, 12)

In [6]:
data.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [9]:
data.isnull().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [12]:
data.describe(include='all').T

count unique                    top  freq  \
Item_Identifier              8523   1559                  FDW13    10   
Item_Weight                7060.0    NaN                    NaN   NaN   
Item_Fat_Content             8523      5                Low Fat  5089   
Item_Visibility            8523.0    NaN                    NaN   NaN   
Item_Type                    8523     16  Fruits and Vegetables  1232   
Item_MRP                   8523.0    NaN                    NaN   NaN   
Outlet_Identifier            8523     10                 OUT027   935   
Outlet_Establishment_Year  8523.0    NaN                    NaN   NaN   
Outlet_Size                  6113      3                 Medium  2793   
Outlet_Location_Type         8523      3                 Tier 3  3350   
Outlet_Type                  8523      4      Supermarket Type1  5577   
Item_Outlet_Sales          8523.0    NaN                    NaN   NaN   

                                  mean          std     min       25%  \
Item_Identifier                    NaN          NaN     NaN       NaN   
Item_Weight                  12.857645     4.643456   4.555   8.77375   
Item_Fat_Content                   NaN          NaN     NaN       NaN   
Item_Visibility               0.066132     0.051598     0.0  0.026989   
Item_Type                          NaN          NaN     NaN       NaN   
Item_MRP                    140.992782    62.275067   31.29   93.8265   
Outlet_Identifier                  NaN          NaN     NaN       NaN   
Outlet_Establishment_Year  1997.831867      8.37176  1985.0    1987.0   
Outlet_Size                        NaN          NaN     NaN       NaN   
Outlet_Location_Type               NaN          NaN     NaN       NaN   
Outlet_Type                        NaN          NaN     NaN       NaN   
Item_Outlet_Sales          2181.288914  1706.499616   33.29  834.2474   

                                50%        75%         max  
Item_Identifier                 NaN        NaN         NaN  
Item_Weight                    12.6      16.85       21.35  
Item_Fat_Content                NaN        NaN         NaN  
Item_Visibility            0.053931   0.094585    0.328391  
Item_Type                       NaN        NaN         NaN  
Item_MRP                   143.0128   185.6437    266.8884  
Outlet_Identifier               NaN        NaN         NaN  
Outlet_Establishment_Year    1999.0     2004.0      2009.0  
Outlet_Size                     NaN        NaN         NaN  
Outlet_Location_Type            NaN        NaN         NaN  
Outlet_Type                     NaN        NaN         NaN  
Item_Outlet_Sales          1794.331  3101.2964  13086.9648

#### Observations
* We Have 8523 Rows and 12 Columns.
* In this data we have 7 Categical Columns and 5 Numerical Columns.
* Categorical Columns: Item_Identifier, Item_Fat_Content, Item_Type, Outlet_Identifier, Outlet_Size, Outlet_Location_Type.
* Numerical Columns: Item_Weight, Item_Visibility, Item_MRP, Outlet_Establishment_Year, Item_Outlet_Sales.
* Item_Weight and Outlet_Size have some null values.
* We can drop Item_identifier and Outler_Identifier becasue these are not very useful while preidction instead of these columns we can use Item_Type and Outlet_Type.
* We have one year column Outlet_Establishment_Year so we can transform this into Age_of_Outlet, where we subtract the current year - Outlet_Establishment_year so with this w can derive the age of Outlets.